### ISSUE TITLE
PasswordInput visibility-toggle buttons not reachable via keyboard nav

### ISSUE THREAD

**Issue by orzechowskid** [mantinedev/mantine#7935]
### Dependencies check up

- [x] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

8.0.2

### What package has an issue?

@mantine/core

### What framework do you use?

React Router

### In which browsers you can reproduce the issue?

Not applicable – issue is not related to the browser

### Describe the bug

the password-visibility toggle button belonging to `<PasswordInput>` components has its `tabIndex` prop set to -1, which allows for programmatic focusing but not keyboard-nav focusing.

https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/PasswordInput/PasswordInput.tsx#L159

https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Global_attributes/tabindex#try_it

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/sandbox/48rr7m

### Possible fix

- set `<ActionIcon>`'s `tabIndex` prop to 0
- have the `<ActionIcon>`'s value for `tabIndex` controlled by a prop on `<PasswordInput>`, possibly with a default value of -1
- remove this prop entirely

### Self-service

- [x] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
Intended behavior to keep the form flow without interruptions. Not planned to be changed.

**Comment by orzechowskid** [comment]
hm.  I understand the desire to preserve the current behavior of this component (even though I may personally disagree with it).  but I'm working on an app which will likely have users who expect keyboard nav, including focus/blur on these specific buttons, to work as expected and I'd really like to use mantine as the UI toolkit - are you really not open to a PR which would make focus traversal behavior configurable (option #2 in the list of possible fixes)?

**Comment by rtivital** [comment]
You can pass props down to the button with `visibilityToggleButtonProps={{ tabIndex: 0 }}` prop.

**Comment by orzechowskid** [comment]
ah, I missed that in the docs.  that will work, thank you (and sorry for the noise)
